examples/templates/java/README.md

```diff
@@ -0,0 +1,254 @@
+
+## Requirements
+
+- Java 8
+- Kafka Bootstrap Server(s) up and running
+- Existing source Kafka topic(s)
+- An existing Pub/Sub destination output topic
+- (Optional) An existing HashiCorp Vault
+- (Optional) A configured secure SSL connection for Kafka
+
+## Getting Started
+
+This section describes what is needed to get the template up and running.
+- Assembling the Uber-JAR
+- Local execution
+- Google Dataflow Template
+  - Set up the environment
+  - Creating the Dataflow Flex Template
+  - Create a Dataflow job to ingest data using the template.
+- Avro format transferring.
+
+## Assembling the Uber-JAR
+
+To run this template the template Java project should be built into
+an Uber JAR file.
+
+Navigate to the Beam folder:
+
+```
+cd /path/to/beam
+```
+
+In order to create Uber JAR with Gradle, [Shadow plugin](https://github.com/johnrengelman/shadow)
+is used. It creates the `shadowJar` task that builds the Uber JAR:
+
+```
+./gradlew -p examples/templates/java/kafka-to-pubsub clean shadowJar
+```
+
+ℹ️ An **Uber JAR** - also known as **fat JAR** - is a single JAR file that contains
+both target package *and* all its dependencies.
+
+The result of the `shadowJar` task execution is a `.jar` file that is generated
+under the `build/libs/` folder in kafka-to-pubsub directory.
+
+## Local execution
+To execute this pipeline locally, specify the parameters:
+- Kafka Bootstrap servers
+- Kafka input topics
+- Pub/Sub output topic
+in the following format:
+```bash
+--bootstrapServers=host:port \
+--inputTopics=your-input-topic \
+--outputTopic=projects/your-project-id/topics/your-topic-pame
+```
+Optionally, to retrieve Kafka credentials for SASL/SCRAM,
+specify a URL to the credentials in HashiCorp Vault and the vault access token:
+```bash
+--secretStoreUrl=http(s)://host:port/path/to/credentials
+--vaultToken=your-token
+```
+Optionally, to configure secure SSL connection between the Beam pipeline and Kafka,
+specify the parameters:
+- A path to a truststore file (it can be a local path or a GCS path, which should start with `gs://`)
+- A path to a keystore file (it can be a local path or a GCS path, which should start with `gs://`)
+- Truststore password
+- Keystore password
+- Key password
+```bash
+--truststorePath=path/to/kafka.truststore.jks
+--keystorePath=path/to/kafka.keystore.jks
+--truststorePassword=your-truststore-password
+--keystorePassword=your-keystore-password
+--keyPassword=your-key-password
+```
+To change the runner, specify:
+```bash
+--runner=YOUR_SELECTED_RUNNER
+```
+See examples/java/README.md for steps and examples to configure different runners.
+
+## Google Dataflow Template
+
+### Setting Up Project Environment
+
+#### Pipeline variables:
+
+```
+PROJECT=id-of-my-project
+BUCKET_NAME=my-bucket
+REGION=my-region
+```
+
+#### Template Metadata Storage Bucket Creation
+
+The Dataflow Flex template has to store its metadata in a bucket in
```

Comment:
I'm not sure it makes sense to discuss Dataflow Flex templates here, we should leave that to https://github.com/GoogleCloudPlatform/DataflowTemplates/pull/176

I think this README should instead discuss how to run _directly_ on Dataflow and/or how to run on some other Beam runners

Reply:
In our case, we describe in Readme how to run the template on Direct runner and with Dataflow. In the readme, we have the "Local execution" part firstly. In this part of README, we discuss Direct runner.

To run the template using Dataflow we need to build it first. Flex template is the better and newest way to do it. The built template could be used in GCP Console or CLI as a "custom template".  Based on this documentation (I added to the readme)

PR which you mentioned targets the repository which contained  Google Provided Templates. But in this PR we planning to give options for Beam users to make custom templates based on our one or using it directly if we cover necessary use-cases.